app/api/groups/get_team_descendants.go

```diff
@@ -0,0 +1,180 @@
+// summary: List team descendants of the group
+// description: Returns all teams (`sType` = "Team") among the descendants of the given group
+//
+//   * The authenticated user should own the parent group.
+// parameters:
+// - name: group_id
+//   in: path
+//   required: true
+//   type: integer
+// - name: from.name
+//   description: Start the page from the team next to the team with `sName` = `from.name` and `ID` = `from.id`
+//                (`from.id` is required when `from.name` is present)
+//   in: query
+//   type: string
+// - name: from.id
+//   description: Start the page from the team next to the team with `sName`=`from.name` and `ID`=`from.id`
+//                (`from.name` is required when from.id is present)
+//   in: query
+//   type: integer
+// - name: sort
+//   in: query
+//   default: [name,id]
+//   type: array
+//   items:
+//     type: string
+//     enum: [name,-name,id,-id]
+// - name: limit
+//   description: Display the first N teams
+//   in: query
+//   type: integer
+//   maximum: 1000
+//   default: 500
+// responses:
+//   "200":
+//     description: OK. Success response with an array of teams
+//     schema:
+//       type: array
+//       items:
+//         "$ref": "#/definitions/teamDescendant"
+//   "400":
+//     "$ref": "#/responses/badRequestResponse"
+//   "401":
+//     "$ref": "#/responses/unauthorizedResponse"
+//   "403":
+//     "$ref": "#/responses/forbiddenResponse"
+//   "500":
+//     "$ref": "#/responses/internalErrorResponse"
+func (srv *Service) getTeamDescendants(w http.ResponseWriter, r *http.Request) service.APIError {
+	user := srv.GetUser(r)
+
+	groupID, err := service.ResolveURLQueryPathInt64Field(r, "group_id")
+	if err != nil {
+		return service.ErrInvalidRequest(err)
+	}
+
+	if apiError := checkThatUserOwnsTheGroup(srv.Store, user, groupID); apiError != service.NoError {
+		return apiError
+	}
+
+	query := srv.Store.Groups().
+		Select("groups.ID, groups.sName, groups.iGrade").
+		Joins(`
+			JOIN groups_ancestors ON groups_ancestors.idGroupChild = groups.ID AND
+				groups_ancestors.idGroupAncestor != groups_ancestors.idGroupChild AND
+				groups_ancestors.idGroupAncestor = ?`, groupID).
+		Where("groups.sType = 'Team'")
+	query = service.NewQueryLimiter().Apply(r, query)
+	query, apiError := service.ApplySortingAndPaging(r, query,
+		map[string]*service.FieldSortingParams{
+			"name": {ColumnName: "groups.sName", FieldType: "string"},
+			"id":   {ColumnName: "groups.ID", FieldType: "int64"}},
+		"name")
+	if apiError != service.NoError {
+		return apiError
+	}
+
+	var result []teamDescendant
+	service.MustNotBeError(query.Scan(&result).Error())
+
+	groupIDs := make([]int64, 0, len(result))
+	resultMap := make(map[int64]*teamDescendant, len(result))
+	for index, groupRow := range result {
+		groupIDs = append(groupIDs, groupRow.ID)
+		resultMap[groupRow.ID] = &result[index]
+		result[index].Members = []teamDescendantMember{}
+	}
+
+	var parentsResult []teamDescendantParent
```

Comment:
If you do it in 3 requests, it assumes you checked it is more efficient than one !?

Reply:
optimized queries to make them more efficient + added an index